apis/v1alpha1/opampbridge_types.go

```diff
@@ -0,0 +1,137 @@
+	// +optional
+	Replicas *int32 `json:"replicas,omitempty"`
+	// SecurityContext will be set as the container security context.
+	// +optional
+	SecurityContext *v1.SecurityContext `json:"securityContext,omitempty"`
+	// PodSecurityContext will be set as the pod security context.
+	// +optional
+	PodSecurityContext *v1.PodSecurityContext `json:"podSecurityContext,omitempty"`
+	// PodAnnotations is the set of annotations that will be attached to
+	// OpAMPBridge pods.
+	// +optional
+	PodAnnotations map[string]string `json:"podAnnotations,omitempty"`
+	// ServiceAccount indicates the name of an existing service account to use with this instance. When set,
+	// the operator will not automatically create a ServiceAccount for the OpAMPBridge.
+	// +optional
+	ServiceAccount string `json:"serviceAccount,omitempty"`
+	// Image indicates the container image to use for the OpAMPBridge.
+	// +optional
+	Image string `json:"image,omitempty"`
+	// UpgradeStrategy represents how the operator will handle upgrades to the CR when a newer version of the operator is deployed
+	// +optional
+	UpgradeStrategy UpgradeStrategy `json:"upgradeStrategy"`
+	// ImagePullPolicy indicates the pull policy to be used for retrieving the container image (Always, Never, IfNotPresent)
+	// +optional
+	ImagePullPolicy v1.PullPolicy `json:"imagePullPolicy,omitempty"`
+	// VolumeMounts represents the mount points to use in the underlying OpAMPBridge deployment(s)
+	// +optional
+	// +listType=atomic
+	VolumeMounts []v1.VolumeMount `json:"volumeMounts,omitempty"`
+	// Ports allows a set of ports to be exposed by the underlying v1.Service.
+	// +optional
+	// +listType=atomic
+	Ports []v1.ServicePort `json:"ports,omitempty"`
+	// ENV vars to set on the OpAMPBridge Pods.
+	// +optional
+	Env []v1.EnvVar `json:"env,omitempty"`
+	// List of sources to populate environment variables on the OpAMPBridge Pods.
+	// +optional
+	EnvFrom []v1.EnvFromSource `json:"envFrom,omitempty"`
+	// Toleration to schedule OpAMPBridge pods.
+	// +optional
+	Tolerations []v1.Toleration `json:"tolerations,omitempty"`
+	// Volumes represents which volumes to use in the underlying OpAMPBridge deployment(s).
+	// +optional
+	// +listType=atomic
+	Volumes []v1.Volume `json:"volumes,omitempty"`
+	// HostNetwork indicates if the pod should run in the host networking namespace.
+	// +optional
+	HostNetwork bool `json:"hostNetwork,omitempty"`
+	// If specified, indicates the pod's priority.
+	// If not specified, the pod priority will be default or zero if there is no
+	// default.
+	// +optional
+	PriorityClassName string `json:"priorityClassName,omitempty"`
+	// If specified, indicates the pod's scheduling constraints
+	// +optional
+	Affinity *v1.Affinity `json:"affinity,omitempty"`
+}
+
+// OpAMPBridgeStatus defines the observed state of OpAMPBridge.
+type OpAMPBridgeStatus struct {
+	// Version of the managed OpAMP Bridge (operand)
+	// +optional
+	Version string `json:"version,omitempty"`
```

Comment:
Maybe we should show something about how many collectors have been created by this bridge? Also maybe a number of pods desired/running?

Reply:
Currently, Status shows Version and OpAMP Server Endpoint. Can we work on other status fields in subsequent PRs ?

Reply:
Sure, that's a fine follow up :)